lms/static/js/ccx/view/schedule_right_container.js

```diff
@@ -0,0 +1,221 @@
+var define = window.define || RequireJS.define;  // jshint ignore:line
+
+define(
+  [
+    'backbone',
+    'jquery',
+    'underscore',
+    'gettext',
+    'text!templates/ccx/underscore/right-container.underscore',
+    'text!templates/ccx/underscore/form.underscore',
+    'js/vendor/timepicker/jquery.timepicker'
+  ],
+  function (
+    Backbone,
+    $,
+    _,
+    gettext,
+    scheduleRightContainerTemplate,
+    scheduleFormTemplate
+  ) {
+    'use strict';
+    return Backbone.View.extend({
+
+      events: {
+        'change select#ccx_chapter': 'onChapterSelect',
+        'change select#ccx_sequential': 'onSubsectionSelect',
+        'change select#ccx_vertical': 'onVerticalSelect',
+        'change .date': 'onDateSelect',
+        'click #add-unit-button': 'addUnitInScheduleTree',
+        'click #add-all': 'showAllUnitsInScheduleTree',
+        'click #save-changes': 'saveSchedule'
+      },
+
+      initialize: function(options) {
+        this.unsavedChanges = options.unsavedChanges;
+
+        // Hidden data will be shown into form,
+        // from there user can add this data to schedule tree
+        this.chapters = this.collection.filterFormData();
+        this.resetFormFlags();
+      },
+
+      render: function() {
+        if (this.chapters) {
+          this.loadRightContainerCCXSchedule();
+        }
+
+        return this;
+      },
+
+      loadRightContainerCCXSchedule: function() {
+        this.$el.html(_.template(scheduleRightContainerTemplate) ({
+          chaptersAvailable: (this.chapters && !_.isEmpty(this.chapters)),
+          unsavedChanges: this.unsavedChanges,
+          form: _.template(scheduleFormTemplate)({
```

Comment:
Since you are passing HTML, use HtmlUtils to wrap this. If you use HtmlUtils.template, the method will automatically return a HtmlSnippet, which is what you want here.

These are recent utility files, developed for the XSS safe template work. There was a link to documentation above.